### PR TITLE
Add new top-level Tracks event quick_action_completed 

### PR DESCRIPTION
**Pull request by mzorz**
The title has it - we're adding a new `quick_action_completed` top-level Tracks event to both platforms so push notifications quick actions can get tracked and processed properly on the Tracks backend. We still keep the other tracks events for `reply`, `approve`, and `like` as these are used for both quick actions _and_ actions happening within the app itself (i.e. with the app in the foreground).

The new event is named `wp(ios|android)_quick_action_completed` and a property with one of 3 possible values depending on the quick action:

```quick_action: (like|approve|reply-to)```

~**A note on naming**: we decided to name it `..._completed` but you'll see the tracks call happens on quick action being triggered rather than on the action having actually been completed. I decided to keep the analytics bump on the same place so the behavior is comparable to what we previously had (declares intent).~
Based on discussion with @stevebaranski, changed the event name to `..._touch` in 87e436f354

**To test:**
#### APPROVE
1. login to the app and make sure the test site you're an admin of has comment moderation enabled.
https://<yourtestblog>.wordpress.com/wp-admin/options-discussion.php
screen shot 2018-10-18 at 08 34 48

(scroll down and save your changes if you need to)
2. on a different app / web, log in with another user and make a comment on a blog post the user in point 1 is an admin of.
3. you should receive a Push Notification with quick actions REPLY/APPROVE. If you don't see the quick actions right away, drag the notification from the bottom edge to make these visible.

screen shot 2018-10-18 at 08 38 01

4. tap on `APPROVE`
5. verify the event exists in tracks for the username you logged in to the app with (step 1) https://mc.a8c.com/tracks/live/?eventname=wpandroid_quick_action_completed&user=&useragent= (you may have to wait just a bit), and the properties include `- quick_action: approve` as seen in 
screen shot 2018-10-18 at 08 41 33

#### REPLY
1. login to the app
2. on a different app / web, log in with another user and make a comment on a blog post the user in point 1 is an admin of.
3. you should receive a Push Notification with quick actions REPLY/APPROVE. If you don't see the quick actions right away, drag the notification from the bottom edge to make these visible.
screen shot 2018-10-18 at 08 38 01

4. tap on `REPLY` and start typing the reply

screen shot 2018-10-18 at 08 40 25
screen shot 2018-10-18 at 08 40 18

5. verify the event exists in tracks for the username you logged in to the app with (step 1) https://mc.a8c.com/tracks/live/?eventname=wpandroid_quick_action_completed&user=&useragent= (you may have to wait just a bit), and the properties include `- quick_action: reply-to` as seen in 
screen shot 2018-10-18 at 09 03 31

#### LIKE
1. login to the app and make sure the test site you're an admin of does NOT have comment moderation enabled.
https://<yourtestblog>.wordpress.com/wp-admin/options-discussion.php

screen shot 2018-10-18 at 08 43 58

(scroll down and save your changes if you need to)
2. on a different app / web, log in with another user and make a comment on a blog post the user in point 1 is an admin of.
3. you should receive a Push Notification with quick actions REPLY/LIKE. If you don't see the quick actions right away, drag the notification from the bottom edge to make these visible.
screen shot 2018-10-18 at 08 46 05

4. tap on `LIKE`
5. verify the event exists in tracks for the username you logged in to the app with (step 1) https://mc.a8c.com/tracks/live/?eventname=wpandroid_quick_action_completed&user=&useragent= (you may have to wait just a bit), and the properties include `- quick_action: like` as seen in 
screen shot 2018-10-18 at 09 03 40

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/push/NotificationsProcessingService.java [hunk 1]
```diff
@@ -596,6 +618,12 @@ private void resetOriginalNotification() {
         }
     }
 
+    private void trackQuickActionCompleted(QuickActionTrackPropertyValue type) {
+        Map<String, String> properties = new HashMap<>(1);
+        properties.put("quick_action", type.toString());
+        AnalyticsTracker.track(AnalyticsTracker.Stat.NOTIFICATION_QUICK_ACTIONS_QUICKACTION_TOUCH, properties);
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/push/NotificationsProcessingService.java]: I would add a call to ` AnalyticsTracker.flush();` here - To be sure all the Analytics data are sent to the remote server.

Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/push/NotificationsProcessingService.java]: 👍 good call, added in 55a0d7c063

File: libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTracker.java [hunk 2]
```diff
@@ -185,6 +185,7 @@
         NOTIFICATION_UNAPPROVED,
         NOTIFICATION_LIKED,
         NOTIFICATION_QUICK_ACTIONS_LIKED,
+        NOTIFICATION_QUICK_ACTIONS_QUICKACTION_TOUCH,
```
Comment (daniloercoli) [inline libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTracker.java]: Should we use `NOTIFICATION_QUICK_ACTIONS_QUICKACTION_TOUCHED`?

File: libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java [hunk 3]
```diff
@@ -715,6 +715,8 @@ public static String getEventNameForStat(AnalyticsTracker.Stat stat) {
             case NOTIFICATION_LIKED:
             case NOTIFICATION_QUICK_ACTIONS_LIKED:
                 return "notifications_comment_liked";
+            case NOTIFICATION_QUICK_ACTIONS_QUICKACTION_TOUCH:
+                return "quick_action_touch";
```
Comment (daniloercoli) [inline libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java]: Same of above, should we use "quick_action_touched"?

File: libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTracker.java (hunk 2 above)
Comment (mzorz) [reply libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTracker.java]: agreed - 64a386c062

File: libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java (hunk 3 above)
Comment (mzorz) [reply libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java]: 👍 addressed in 64a386c062 as well

**Comment by daniloercoli** [comment]
LGTM!

**Pull request merged**